|ウィスコンシン乳がんデータセット|
|:-|

乳がんの腫瘍を計測したもので，それぞれの腫瘍に，害のない腫瘍を意味する「良性（benign）」か，悪性の腫瘍を意味する「（malignant）」かのラベルが付けられている．ここでのタスクは，組織の測定結果から腫瘍が悪性かどうかを予測するように学習することである．

In [ ]:
from sklearn.datasets import load_breast_cancer
cancer = load_breast_cancer()
print(cancer)
print("cancer.keys(): \n{}".format(cancer.keys()))

{'data': array([[1.799e+01, 1.038e+01, 1.228e+02, ..., 2.654e-01, 4.601e-01,
        1.189e-01],
       [2.057e+01, 1.777e+01, 1.329e+02, ..., 1.860e-01, 2.750e-01,
        8.902e-02],
       [1.969e+01, 2.125e+01, 1.300e+02, ..., 2.430e-01, 3.613e-01,
        8.758e-02],
       ...,
       [1.660e+01, 2.808e+01, 1.083e+02, ..., 1.418e-01, 2.218e-01,
        7.820e-02],
       [2.060e+01, 2.933e+01, 1.401e+02, ..., 2.650e-01, 4.087e-01,
        1.240e-01],
       [7.760e+00, 2.454e+01, 4.792e+01, ..., 0.000e+00, 2.871e-01,
        7.039e-02]]), 'target': array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0,
       1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0,
       1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 1,
       1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0,
 

In [ ]:
print("Shape of cancer data: {}".format(cancer.data.shape)) #このデータセットは，30の特徴量を持つ569おデータポイントで構成されている．
print(cancer.data)#このデータセットは，30の特徴量を持つ569のデータポイントで構成されている．
print(len(cancer.data)) # 569
print(len(cancer.data[0])) # 30

Shape of cancer data: (569, 30)
[[1.799e+01 1.038e+01 1.228e+02 ... 2.654e-01 4.601e-01 1.189e-01]
 [2.057e+01 1.777e+01 1.329e+02 ... 1.860e-01 2.750e-01 8.902e-02]
 [1.969e+01 2.125e+01 1.300e+02 ... 2.430e-01 3.613e-01 8.758e-02]
 ...
 [1.660e+01 2.808e+01 1.083e+02 ... 1.418e-01 2.218e-01 7.820e-02]
 [2.060e+01 2.933e+01 1.401e+02 ... 2.650e-01 4.087e-01 1.240e-01]
 [7.760e+00 2.454e+01 4.792e+01 ... 0.000e+00 2.871e-01 7.039e-02]]
569
30


このデータセットは，569のデータポイントと30の特徴量で構成されている．

In [ ]:
print("Sample counts per class:\n{}".format({n: v for n, v in zip(cancer.target_names, np.bincount(cancer.target))})) 
#悪性が212良性が357
# np.bincount(cancer.target)は，cancer.target内の0と1の要素数を数えてこの順番に要素数を出力する

Sample counts per class:
{'malignant': 212, 'benign': 357}


569のデータポイントのうち，212が悪性で357が良性である．

In [ ]:
#個々の特徴量の意味を示す記述
print("Feature names:\n{}".format(cancer.feature_names))

Feature names:
['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension']


In [ ]:
print(len(cancer.feature_names)) # 特徴量の数を数える

30


実世界の回帰データセットとして，boston_housingデータセットを用いる．このデータセットを用いるタスクは，1970年代のボストン近郊の住宅地の住宅価格の中央値を，犯罪率，チャールズ川からの距離，高速道路への利便性などから予測するものだ，このデータセットには，13の特徴量を持つ506のデータポイントが含まれる．

In [ ]:
from sklearn.datasets import fetch_california_housing
california = fetch_california_housing()
print("Data shape: {}".format(california.data.shape))

Data shape: (20640, 8)


<div style="background-color:#02383b; padding: 10px; border-radius: 5px;">

[`fetch_california_housing`](command:_github.copilot.openSymbolFromReferences?%5B%22fetch_california_housing%22%2C%5B%7B%22uri%22%3A%7B%22%24mid%22%3A1%2C%22fsPath%22%3A%22c%3A%5C%5CUsers%5C%5Cseiji%5C%5COneDrive%20-%20Kyoto%20University%20(1)%5C%5C1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%5C%5Cprogramming%5C%5Cpython%5C%5CPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%5C%5Cpython_macinelearning_oreilly%5C%5C2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22_sep%22%3A1%2C%22external%22%3A%22vscode-notebook-cell%3A%2Fc%253A%2FUsers%2Fseiji%2FOneDrive%2520-%2520Kyoto%2520University%2520%25281%2529%2F1.Kyoto.Univ%25E3%2581%259D%25E3%2581%25AE%25E4%25BB%2596%2Fprogramming%2Fpython%2FPython%25E3%2581%25A7%25E5%25A7%258B%25E3%2582%2581%25E3%2582%258B%25E6%25A9%259F%25E6%25A2%25B0%25E5%25AD%25A6%25E7%25BF%2592%2Fpython_macinelearning_oreilly%2F2%25E7%25AB%25A0_%25E6%2595%2599%25E5%25B8%25AB%25E3%2581%2582%25E3%2582%258A%25E5%25AD%25A6%25E7%25BF%2592.ipynb%23X30sZmlsZQ%253D%253D%22%2C%22path%22%3A%22%2Fc%3A%2FUsers%2Fseiji%2FOneDrive%20-%20Kyoto%20University%20(1)%2F1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%2Fprogramming%2Fpython%2FPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%2Fpython_macinelearning_oreilly%2F2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22scheme%22%3A%22vscode-notebook-cell%22%2C%22fragment%22%3A%22X30sZmlsZQ%3D%3D%22%7D%2C%22pos%22%3A%7B%22line%22%3A0%2C%22character%22%3A29%7D%7D%5D%5D "Go to definition")は、Scikit-learnライブラリの`sklearn.datasets`モジュールに含まれる関数で、カリフォルニア州の住宅データセットを取得するために使用されます。このデータセットは、回帰分析の練習やテストに広く使用されます。

### 計画
1. [`fetch_california_housing`](command:_github.copilot.openSymbolFromReferences?%5B%22fetch_california_housing%22%2C%5B%7B%22uri%22%3A%7B%22%24mid%22%3A1%2C%22fsPath%22%3A%22c%3A%5C%5CUsers%5C%5Cseiji%5C%5COneDrive%20-%20Kyoto%20University%20(1)%5C%5C1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%5C%5Cprogramming%5C%5Cpython%5C%5CPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%5C%5Cpython_macinelearning_oreilly%5C%5C2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22_sep%22%3A1%2C%22external%22%3A%22vscode-notebook-cell%3A%2Fc%253A%2FUsers%2Fseiji%2FOneDrive%2520-%2520Kyoto%2520University%2520%25281%2529%2F1.Kyoto.Univ%25E3%2581%259D%25E3%2581%25AE%25E4%25BB%2596%2Fprogramming%2Fpython%2FPython%25E3%2581%25A7%25E5%25A7%258B%25E3%2582%2581%25E3%2582%258B%25E6%25A9%259F%25E6%25A2%25B0%25E5%25AD%25A6%25E7%25BF%2592%2Fpython_macinelearning_oreilly%2F2%25E7%25AB%25A0_%25E6%2595%2599%25E5%25B8%25AB%25E3%2581%2582%25E3%2582%258A%25E5%25AD%25A6%25E7%25BF%2592.ipynb%23X30sZmlsZQ%253D%253D%22%2C%22path%22%3A%22%2Fc%3A%2FUsers%2Fseiji%2FOneDrive%20-%20Kyoto%20University%20(1)%2F1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%2Fprogramming%2Fpython%2FPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%2Fpython_macinelearning_oreilly%2F2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22scheme%22%3A%22vscode-notebook-cell%22%2C%22fragment%22%3A%22X30sZmlsZQ%3D%3D%22%7D%2C%22pos%22%3A%7B%22line%22%3A0%2C%22character%22%3A29%7D%7D%5D%5D "Go to definition")の概要を説明。
2. データセットの内容を説明。
3. 使用例を示す。

### 概要
- [`fetch_california_housing`](command:_github.copilot.openSymbolFromReferences?%5B%22fetch_california_housing%22%2C%5B%7B%22uri%22%3A%7B%22%24mid%22%3A1%2C%22fsPath%22%3A%22c%3A%5C%5CUsers%5C%5Cseiji%5C%5COneDrive%20-%20Kyoto%20University%20(1)%5C%5C1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%5C%5Cprogramming%5C%5Cpython%5C%5CPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%5C%5Cpython_macinelearning_oreilly%5C%5C2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22_sep%22%3A1%2C%22external%22%3A%22vscode-notebook-cell%3A%2Fc%253A%2FUsers%2Fseiji%2FOneDrive%2520-%2520Kyoto%2520University%2520%25281%2529%2F1.Kyoto.Univ%25E3%2581%259D%25E3%2581%25AE%25E4%25BB%2596%2Fprogramming%2Fpython%2FPython%25E3%2581%25A7%25E5%25A7%258B%25E3%2582%2581%25E3%2582%258B%25E6%25A9%259F%25E6%25A2%25B0%25E5%25AD%25A6%25E7%25BF%2592%2Fpython_macinelearning_oreilly%2F2%25E7%25AB%25A0_%25E6%2595%2599%25E5%25B8%25AB%25E3%2581%2582%25E3%2582%258A%25E5%25AD%25A6%25E7%25BF%2592.ipynb%23X30sZmlsZQ%253D%253D%22%2C%22path%22%3A%22%2Fc%3A%2FUsers%2Fseiji%2FOneDrive%20-%20Kyoto%20University%20(1)%2F1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%2Fprogramming%2Fpython%2FPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%2Fpython_macinelearning_oreilly%2F2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22scheme%22%3A%22vscode-notebook-cell%22%2C%22fragment%22%3A%22X30sZmlsZQ%3D%3D%22%7D%2C%22pos%22%3A%7B%22line%22%3A0%2C%22character%22%3A29%7D%7D%5D%5D "Go to definition")は、カリフォルニア州の住宅価格に関するデータセットをロードします。
- データセットには、1990年のカリフォルニア州の各地区の住宅価格と、それに関連する8つの特徴量が含まれています。

### データセットの内容
- 特徴量（8つの特徴量）：
  - `MedInc`: 地区の世帯の中央値収入
  - `HouseAge`: 地区の住宅の中央値の築年数
  - `AveRooms`: 1世帯あたりの平均部屋数
  - `AveBedrms`: 1世帯あたりの平均寝室数
  - `Population`: 地区の人口
  - `AveOccup`: 1世帯あたりの平均居住者数
  - `Latitude`: 地区の緯度
  - `Longitude`: 地区の経度
- 目標変数：
  - `MedHouseVal`: 地区の住宅価格の中央値

### 使用例

</div>

In [ ]:
from sklearn.datasets import fetch_california_housing

# データセットをロード
housing = fetch_california_housing()

# 特徴量と目標変数を取得
X = housing.data
y = housing.target

# データの概要を表示
print(housing.DESCR)
print(X.shape)  # 特徴量の形状
print(y.shape)  # 目標変数の形状

.. _california_housing_dataset:

California Housing dataset
--------------------------

**Data Set Characteristics:**

:Number of Instances: 20640

:Number of Attributes: 8 numeric, predictive attributes and the target

:Attribute Information:
    - MedInc        median income in block group
    - HouseAge      median house age in block group
    - AveRooms      average number of rooms per household
    - AveBedrms     average number of bedrooms per household
    - Population    block group population
    - AveOccup      average number of household members
    - Latitude      block group latitude
    - Longitude     block group longitude

:Missing Attribute Values: None

This dataset was obtained from the StatLib repository.
https://www.dcc.fc.up.pt/~ltorgo/Regression/cal_housing.html

The target variable is the median house value for California districts,
expressed in hundreds of thousands of dollars ($100,000).

This dataset was derived from the 1990 U.S. census, using one row per ce

<div style="background-color:#02023b; padding: 10px; border-radius: 5px;">

このコードでは、[`fetch_california_housing`](command:_github.copilot.openSymbolFromReferences?%5B%22fetch_california_housing%22%2C%5B%7B%22uri%22%3A%7B%22%24mid%22%3A1%2C%22fsPath%22%3A%22c%3A%5C%5CUsers%5C%5Cseiji%5C%5COneDrive%20-%20Kyoto%20University%20(1)%5C%5C1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%5C%5Cprogramming%5C%5Cpython%5C%5CPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%5C%5Cpython_macinelearning_oreilly%5C%5C2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22_sep%22%3A1%2C%22external%22%3A%22vscode-notebook-cell%3A%2Fc%253A%2FUsers%2Fseiji%2FOneDrive%2520-%2520Kyoto%2520University%2520%25281%2529%2F1.Kyoto.Univ%25E3%2581%259D%25E3%2581%25AE%25E4%25BB%2596%2Fprogramming%2Fpython%2FPython%25E3%2581%25A7%25E5%25A7%258B%25E3%2582%2581%25E3%2582%258B%25E6%25A9%259F%25E6%25A2%25B0%25E5%25AD%25A6%25E7%25BF%2592%2Fpython_macinelearning_oreilly%2F2%25E7%25AB%25A0_%25E6%2595%2599%25E5%25B8%25AB%25E3%2581%2582%25E3%2582%258A%25E5%25AD%25A6%25E7%25BF%2592.ipynb%23X30sZmlsZQ%253D%253D%22%2C%22path%22%3A%22%2Fc%3A%2FUsers%2Fseiji%2FOneDrive%20-%20Kyoto%20University%20(1)%2F1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%2Fprogramming%2Fpython%2FPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%2Fpython_macinelearning_oreilly%2F2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22scheme%22%3A%22vscode-notebook-cell%22%2C%22fragment%22%3A%22X30sZmlsZQ%3D%3D%22%7D%2C%22pos%22%3A%7B%22line%22%3A0%2C%22character%22%3A29%7D%7D%5D%5D "Go to definition")関数を使用してデータセットをロードし、特徴量と目標変数を取得しています。また、データセットの概要と形状を表示しています。

</div>

In [ ]:
from sklearn.datasets import fetch_california_housing

# データセットをロード
housing = fetch_california_housing()

# 特徴量と目標変数を取得
X = housing.data
y = housing.target

# データの概要を表示
print(housing.DESCR)
print(X.shape)  # 特徴量の形状
print(y.shape)  # 目標変数の形状

.. _california_housing_dataset:

California Housing dataset
--------------------------

**Data Set Characteristics:**

:Number of Instances: 20640

:Number of Attributes: 8 numeric, predictive attributes and the target

:Attribute Information:
    - MedInc        median income in block group
    - HouseAge      median house age in block group
    - AveRooms      average number of rooms per household
    - AveBedrms     average number of bedrooms per household
    - Population    block group population
    - AveOccup      average number of household members
    - Latitude      block group latitude
    - Longitude     block group longitude

:Missing Attribute Values: None

This dataset was obtained from the StatLib repository.
https://www.dcc.fc.up.pt/~ltorgo/Regression/cal_housing.html

The target variable is the median house value for California districts,
expressed in hundreds of thousands of dollars ($100,000).

This dataset was derived from the 1990 U.S. census, using one row per ce

<div style="background-color:#02023b; padding: 10px; border-radius: 5px;">

このコードでは、[`fetch_california_housing`](command:_github.copilot.openSymbolFromReferences?%5B%22fetch_california_housing%22%2C%5B%7B%22uri%22%3A%7B%22%24mid%22%3A1%2C%22fsPath%22%3A%22c%3A%5C%5CUsers%5C%5Cseiji%5C%5COneDrive%20-%20Kyoto%20University%20(1)%5C%5C1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%5C%5Cprogramming%5C%5Cpython%5C%5CPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%5C%5Cpython_macinelearning_oreilly%5C%5C2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22_sep%22%3A1%2C%22external%22%3A%22vscode-notebook-cell%3A%2Fc%253A%2FUsers%2Fseiji%2FOneDrive%2520-%2520Kyoto%2520University%2520%25281%2529%2F1.Kyoto.Univ%25E3%2581%259D%25E3%2581%25AE%25E4%25BB%2596%2Fprogramming%2Fpython%2FPython%25E3%2581%25A7%25E5%25A7%258B%25E3%2582%2581%25E3%2582%258B%25E6%25A9%259F%25E6%25A2%25B0%25E5%25AD%25A6%25E7%25BF%2592%2Fpython_macinelearning_oreilly%2F2%25E7%25AB%25A0_%25E6%2595%2599%25E5%25B8%25AB%25E3%2581%2582%25E3%2582%258A%25E5%25AD%25A6%25E7%25BF%2592.ipynb%23X30sZmlsZQ%253D%253D%22%2C%22path%22%3A%22%2Fc%3A%2FUsers%2Fseiji%2FOneDrive%20-%20Kyoto%20University%20(1)%2F1.Kyoto.Univ%E3%81%9D%E3%81%AE%E4%BB%96%2Fprogramming%2Fpython%2FPython%E3%81%A7%E5%A7%8B%E3%82%81%E3%82%8B%E6%A9%9F%E6%A2%B0%E5%AD%A6%E7%BF%92%2Fpython_macinelearning_oreilly%2F2%E7%AB%A0_%E6%95%99%E5%B8%AB%E3%81%82%E3%82%8A%E5%AD%A6%E7%BF%92.ipynb%22%2C%22scheme%22%3A%22vscode-notebook-cell%22%2C%22fragment%22%3A%22X30sZmlsZQ%3D%3D%22%7D%2C%22pos%22%3A%7B%22line%22%3A0%2C%22character%22%3A29%7D%7D%5D%5D "Go to definition")関数を使用してデータセットをロードし、特徴量と目標変数を取得しています。また、データセットの概要と形状を表示しています。

</div>

このデータセットは，20640のデータポイントと8の特徴量で構成されている．

In [ ]:
X, y = mglearn.datasets.load_extended_boston()
print("X.shape: {}".format(X.shape))

X.shape: (506, 104)
